src/OctoshiftCLI.Tests/GithubApiTests.cs

```diff
@@ -2255,6 +2255,31 @@ public async Task StartGitArchiveGeneration_Returns_The_Initiated_Migration_Id()
             actualMigrationId.Should().Be(expectedMigrationId);
         }
 
+        [Fact]
+        public async Task StartGitArchiveGeneration_Throws_Octoshift_CLI_Exception_When_Blob_Storage_Settings_Are_Not_Set()
+        {
+            // Arrange
+            const string url = $"https://api.github.com/orgs/{GITHUB_ORG}/migrations";
+            var payload = new
+            {
+                repositories = new[] { GITHUB_REPO },
+                exclude_metadata = true
+            };
+            var exception_message = "Before you can start a migration, you must configure blob storage settings in your management console.";
+
+            _githubClientMock
+                .Setup(m => m.PostAsync(url, It.Is<object>(x => x.ToJson() == payload.ToJson()), null))
+                    .ThrowsAsync(new HttpRequestException(exception_message, null, HttpStatusCode.BadGateway));
+
+            // Act
+            Func<Task> act = async () => await _githubApi.StartGitArchiveGeneration(GITHUB_ORG, GITHUB_REPO);
+
+            // Assert
+            await act.Should().ThrowExactlyAsync<OctoshiftCliException>()
+                .WithMessage(exception_message);
```

Comment:
The other unit tests in here use a slightly different pattern to test for exceptions (the way you did it here actually is probably nicer, but we should probably be consistent unless we want to change it everywhere).

```suggestion
            await _githubApi.Invoking(api => api.StartGitArchiveGeneration(GITHUB_ORG, GITHUB_REPO))
                .Should()
                .ThrowExactlyAsync<OctoshiftCliException>()
                .WithMessage(exception_message);
```